includes/class-amp-theme-support.php

```diff
@@ -829,7 +828,21 @@ protected static function ensure_required_markup( DOMDocument $dom ) {
 			) );
 			$head->insertBefore( $meta_viewport, $meta_charset->nextSibling );
 		}
-
+		// Prevent schema.org duplicates.
+		$schema_org_script = null;
+		foreach ( $head->getElementsByTagName( 'script' ) as $script ) {
+			if ( 'application/ld+json' === $script->getAttribute( 'type' ) && preg_match( '/{"@context":"https?:[\\\]\/[\\\]\/schema\.org/', $script->nodeValue ) ) {
+				$schema_org_script = $script->nodeValue;
+				break;
+			}
+		}
+		if ( ! $schema_org_script ) {
+			$script = $dom->createElement( 'script', wp_json_encode( amp_get_schemaorg_metadata() ) );
+			AMP_DOM_Utils::add_attributes_to_node( $script, array(
```

Comment:
Since you're using `$dom->createElement()` here then I think it makes more sense to just do `$script->setAttribute( 'type', 'application/ld+json' )`

Reply:
Also true, corrected it.